Add re-testing solutions documentation (#448)
exercism/docs: building/tracks/README.md

|           ├── debug.md
|           ├── help.md
|           └── tests.md
└── config.json
</pre>

## Re-testing solutions

When you merge a track PR that touches an exercise, it triggers the latest published iteration of students' solutions to be re-tested.
For popular exercises, this is a _very_ expensive operation (70,000 test runs for Python Hello World as an extreme!).

> We've added some further rules today to try and avoid this behaviour. I'll explain the logic in both directions:

Reviewer: ```suggestion
We encourage you to try and doing this unnecessarily.
```

Solutions **will not** be retested if:

- You add `[no important files changed]` into the PR's commit body.

> - You only touch docs, hints, `.meta`, or other files that users don't interact with.

Reviewer 2: ```suggestion
Solutions **will not** be retested if the merged commit either:

- only touches docs, hints, `.meta`, or other files that users don't interact with
- or contains `[no important files changed]` in the commit body.
```

Solutions **will** be re-tested if:

1. The PR was merged **without** adding the text `[no important files changed]` into the PR's commit body.

> 2. The PR touches one of the following files for an exercise (as specified in its `.meta/config.json` file):

Reviewer 2: ```suggestion
Solutions **will** be re-tested if the merged commit both:

- lacks `[no important files changed]` in the commit body
- and touches one of the following files for an exercise (as specified in its `.meta/config.json` file):
```

> 

Reviewer: ```suggestion
```

> - test files
> - editor files
> - invalidator files

Reviewer: ```suggestion
  - test files
  - editor files
  - invalidator files
```
